guardian/VaultDoor
# Fix/bring back download

## What does this change?
Bring back the "Download" button in the File Search view. 

In doing so, I've converted a few components into Typescript from Javascript and used React's Context mechanism to propagate the currently selected vault ID within the File Search view.

## How to test
Go to "File Search", any current vault and click a file. You should see the Download button appear where it used to.

## How can we measure success?
Able to do per-file downloads again

## Images
<img width="1536" alt="Screenshot 2021-06-09 at 17 30 12" src="https://user-images.githubusercontent.com/12482441/121393626-72748800-c948-11eb-8904-4c1c549b6ad5.png">
